deno/fhirchat-demo/islands/ThoughtsActionsIsland.tsx

```diff
@@ -0,0 +1,72 @@
+import { useContext } from "preact/hooks";
+import { MOCK_THOUGHT_ACTIONS } from "../constants/mock_thought_actions.ts";
+import { AppState } from "../hooks/appContext.ts";
+
+export default function ThoughtsActionsIsland() {
+  const { thoughtActionPanelOpen, setThoughtsActionsPanel } = useContext(
+    AppState,
+  );
+
+  return (
+    <aside
+      class={`menu slide_out_container ${
+        thoughtActionPanelOpen.value ? "is-visible" : "is-hidden"
+      }`}
+    >
+      <div class="close_button control is-flex is-align-items-center mb-3">
+        <span
+          class="icon is-medium"
+          onClick={() => setThoughtsActionsPanel(false)}
+        >
+          <img src={"../images/close.svg"} alt="close" />
+        </span>
+        CLOSE
+      </div>
+      <p class="slide_out_title mb-2">Thoughts and Actions</p>
+      <p class="slide_out_description mb-6">
+        The steps and actions taken by the FhirChat agent to generate the
+        answer.
+      </p>
+
+      <p class="slide_out_question mb-2">
+        You wrote: How many female patients were born before 1977?
+      </p>
+
+      {MOCK_THOUGHT_ACTIONS.map((item) => (
```

Comment:
The mock should stay hidden from the frontend, as the backend is already providing the mocked answers.
Meaning that there is likely one missing _conversation state_ trick in the recipe.
Where the received response logs, in the `aiConversationContext` should be kept/reused/transformed to be usable within this sliding panel as well.

Reply:
Will these thoughts be present in the conversation object?

Reply:
I decided to go with the another signal `storedThoughts` and keep them separate, but could be swayed if there is a good reason to keep everything in `conversation`